server/src/main/java/org/elasticsearch/search/aggregations/AdaptingAggregator.java

```diff
@@ -101,7 +101,12 @@ public final void postCollection() throws IOException {
 
     @Override
     public final InternalAggregation buildEmptyAggregation() {
-        return adapt(delegate.buildEmptyAggregation());
+        try {
+            return adapt(delegate.buildEmptyAggregation());
+        } catch (IOException e) {
```

Comment:
I wonder moving this inside adapt would be a better way of dealing with it.

Reply:
Yeah. I'm not super sure either. I liked the idea of it being able to throw but I can move it and see if we find this another time.